### PR TITLE
Add ability to get all notifications and clear specific ones

### PR DESCRIPTION
**Pull request by duncan-c**

### CODE REVIEW

File: src/android/FirebasePlugin.java [hunk 1]
```diff
@@ -183,7 +183,7 @@ public void run() {
                     authStateListener = new AuthStateListener();
                     FirebaseAuth.getInstance().addAuthStateListener(authStateListener);
 
-                    firestore = FirebaseFirestore.getInstance();
```
Comment (Aidenjl193) [inline src/android/FirebasePlugin.java:186]: Nice

**Review by Aidenjl193 — APPROVED** [review]
I'll take your word for it

File: src/android/FirebasePlugin.java [hunk 2]
```diff
@@ -1912,6 +1912,50 @@ public void run() {
...
+                }
+            }
+        });
+    }
+
+    public void getAllNotifications(final CallbackContext callbackContext) {
+        cordova.getThreadPool().execute(new Runnable() {
+            public void run() {
+                // only call on Android M and above
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (CalumMurray) [inline src/android/FirebasePlugin.java:1939]: Does this mean on older Androids we'll never clear some notifications (leaving them to be manually cleared only)?

File: www/firebase.js [hunk 3]
```diff
@@ -204,6 +204,24 @@ exports.clearAllNotifications = function (success, error) {
   exec(success, error, "FirebasePlugin", "clearAllNotifications", []);
 };
 
+exports.clearNotification = function (id, tag, success, error) {
+    var args = [id];
+    // "tag" is an optional arg
+    if (tag) {
+        if (typeof tag === 'function') {
```
Comment (CalumMurray) [inline www/firebase.js:211]: I'm confused, can we not pass null or something here and leave success/error callbacks in the right order?

**Review by CalumMurray — APPROVED** [review]
Great work! Can't be certain I've not missed anything, but looks logical to me...

File: src/android/FirebasePlugin.java (hunk 2 above)
Comment (duncan-c) [reply src/android/FirebasePlugin.java:1939]: Yeah, although really old versions (v6), so serves them right

File: www/firebase.js (hunk 3 above)
Comment (duncan-c) [reply www/firebase.js:211]: Could do, but we'd always need to remember that on the app which is risky. This approach isn't pretty but it's how the plugin does it in other places so I figured I'd stick with it.

**Pull request merged**